Add addTargets function for table panel

jdbaldry opened the pull request.

roidelapluie commented:
Hi,

Can you implement it differently as I fear this will have a huge perf impact with many targets?

jdbaldry commented:
Hey! 

I'm still learning jsonnet as a language but am interested in finding a more performant solution. Do you have any recommendations for an alternative implementation as I just took this from from the graph panel. 

I have written a jsonnet snippet that I can use for testing the time of addTargets:
```
// add-panels-test.jsonnet
local grafana = import "grafonnet/grafana.libsonnet";

local dashboard = grafana.dashboard;
local prometheus = grafana.prometheus;
local tablePanel = grafana.tablePanel;


local tests = {
  manyTargets(numTargets):: [
    prometheus.target('up') for i in std.range(1, numTargets)
  ],

  addManyPanelsWithManyTargets(numPanels, numTargets):: [
    tablePanel.new('test').addTargets(self.manyTargets(numTargets)) {
      gridPos: { x: 0, y: 0, h: 8, w: 24 },
    } for i in std.range(1, numPanels)
  ],

};

dashboard.new('add panels test', editable=true, schemaVersion=16)
.addPanel(tablePanel.new('test'), gridPos={x: 0, y: 0, h: 8, w: 24}) + {
  local panels = self.panels,

  panels+: tests.addManyPanelsWithManyTargets(numPanels=200, numTargets=50)
}
```

On my system a test with 200 panels and 50 targets takes just over five seconds:
`time jsonnet add-panels-test.jsonnet &>/dev/null`
```
real	0m4.938s
user	0m4.879s
```

roidelapluie commented:
Look what we do for addPanels.

jdbaldry commented:
Thanks for the pointer. I've updated the code to use the same method from addPanels and updated formatting. All the tests still pass and it is much faster than using foldl.

Does the graph panel want similar changes, perhaps targets should be split into their own module since multiple panels use them?
